damus/Views/QRCodeView.swift

```diff
@@ -23,80 +37,198 @@ struct QRCodeView: View {
+            Spacer()
+        }
+    }
+    
+    func search_changed(_ new: String) {
+        var str = new
+        guard str.count != 0 else {
+            return
+        }
         
-            VStack(alignment: .center) {
-                
-                let profile = damus_state.profiles.lookup(id: pubkey)
-                
-                if (damus_state.profiles.lookup(id: pubkey)?.picture) != nil {
-                    ProfilePicView(pubkey: pubkey, size: 90.0, highlight: .custom(DamusColors.white, 4.0), profiles: damus_state.profiles)
-                        .padding(.top, 50)
-                } else {
-                    Image(systemName: "person.fill")
-                        .font(.system(size: 60))
-                        .foregroundColor(DamusColors.white)
-                        .padding(.top, 50)
-                }
-                
-                if let display_name = profile?.display_name {
-                    Text(display_name)
-                        .foregroundColor(DamusColors.white)
-                        .font(.system(size: 24, weight: .heavy))
-                }
-                if let name = profile?.name {
-                    Text("@" + name)
-                        .foregroundColor(DamusColors.white)
-                        .font(.body)
+        if str.hasPrefix("nostr:") {
+            str.removeFirst("nostr:".count)
+        }
+        
+        if let _ = hex_decode(str), str.count == 64 {
+            self.scanResult = .hex(str)
+            return
+        }
+        
+        if str.starts(with: "npub") {
+            if let _ = try? bech32_decode(str) {
+                self.scanResult = .profile(str)
+                return
+            }
+        }
+    }
+    
+    var QRCameraView: some View {
+        VStack(alignment: .center) {
+            Text("Scan a user's pubkey")
+                .padding(.top, 50)
+                .font(.system(size: 24, weight: .heavy))
+            
+            Spacer()
+
+            CodeScannerView(codeTypes: [.qr], scanMode: .continuous, simulatedData: "npub1k92qsr95jcumkpu6dffurkvwwycwa2euvx4fthv78ru7gqqz0nrs2ngfwd", shouldVibrateOnSuccess: false) { response in
+                switch response {
+                    case .success(let result):
+                        search_changed(result.string)
+                        print("Found code: \(result.string)")
+                        switch scanResult {
+                            case .profile(let prof):
+                                if (currentScannedCode != prof) {
+                                    generator.impactOccurred()
+                                    cameraAnimate()
+                                }
+                                currentScannedCode = prof
+                                let decoded = try? bech32_decode(prof)
+                                let hex = hex_encode(decoded!.data)
+                                profile = damus_state.profiles.lookup(id: hex)
```

Comment:
The proper way to do this is to make a pubkey state variable instead of the actual view